name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/editor/ClientInput.java

```diff
@@ -86,14 +89,17 @@ public class ClientInput
     @Inject
     private IEclipseContext context;
 
+    private MPart part;
+
     @Inject
     @Preference
     private IEclipsePreferences preferences;
 
-    /* protected */ ClientInput(String label, File clientFile)
+    /* protected */ ClientInput(String label, File clientFile, MPart part)
```

Comment:
Keep in mind that there can be many parts for one client input.
A part represents a window. There can be many windows open, but the file is only opened once.
Therefore I think the part should not be added to the ClientInput.

Reply:
Agree. As written in https://github.com/portfolio-performance/portfolio/pull/3510#issuecomment-1686881377. When there can be multiple parts, then the current migration logic does not make any sense.